### ISSUE TITLE
No communication between Homey and Bluesound

### ISSUE THREAD

**Issue by ferdw** [roelofoomen/com.bluesound#1]
Seems that source has fixed IP and MAC for developping.

**Comment by jghaanstra** [comment]
Thanx for submitting this issue. I accidently published my development version which contains mock data (I dont own a Bluesound device myself).

I have just pushed another release to the app store. You can already try it out doing a CLI install.

https://github.com/jghaanstra/com.bluesound/releases/tag/1.0.2

**Comment by ferdw** [comment]
1.0.2 is installed but can not connect:
![2018-01-29 19_13_34-zones devices - homey](https://user-images.githubusercontent.com/10448491/35526702-a3656566-0528-11e8-9504-9101c1843746.png)

I can ping the Bluesound from the Node.js

**Comment by jghaanstra** [comment]
I have created a debug version which contains logging that will hopefully tell me what is going on. Could you run this from the command line (`athom project --run`)?

https://github.com/jghaanstra/com.bluesound/tree/debug

**Comment by ferdw** [comment]
2018-01-29 19:56:55 [log] [BluesoundApp] Initializing Bluesound app ...
testing connection
testing connection not succesfull, error is:
ReferenceError: rp is not defined
    at /lib/util.js:25:9
    at new Promise (<anonymous>)
    at Object.exports.sendCommand (/lib/util.js:18:12)
    at EventEmitter.<anonymous> (/drivers/bluesound/driver.js:18:18)
    at emitTwo (events.js:126:13)
    at EventEmitter.emit (events.js:214:7)
    at i.ready (/opt/homey-client/system/manager/ManagerApps/bootstrap/sdk/v2/manager/drivers.js:1:2094)
    at BluesoundDriver.ready (/opt/homey-client/system/manager/ManagerApps/bootstrap/sdk/v2/lib/Driver.js:1:4331)
    at Homey.ready (/opt/homey-client/system/manager/ManagerApps/bootstrap/sdk/v2/manager/drivers.js:1:2007)
    at SDK._onReady (/opt/homey-client/system/manager/ManagerApps/bootstrap/sdk/v2/lib/SDK.js:1:985)

**Comment by jghaanstra** [comment]
Thanx, could you try again.

https://github.com/jghaanstra/com.bluesound/tree/debug

**Comment by ferdw** [comment]
And with result:

2018-01-29 20:14:12 [log] [BluesoundApp] Initializing Bluesound app ...
testing connection
connection test succesfull, response.body is:
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<SyncStatus icon="/images/players/N110_nt.png" volume="32" modelName="NODE 2" name="Bluesound Woonkamer" model="N110" brand="Bluesound" etag="236" schemaVersion="21" syncStat="236" id="192.168.0.110:11000" mac="xx:xx:xx:xx:ED:A5">
</SyncStatus>

testing connection succesfull, response is:
{ SyncStatus:
   { '$':
      { icon: '/images/players/N110_nt.png',
        volume: '32',
        modelName: 'NODE 2',
        name: 'Bluesound Woonkamer',
        model: 'N110',
        brand: 'Bluesound',
        etag: '236',
        schemaVersion: '21',
        syncStat: '236',
        id: '192.168.0.110:11000',
        mac: 'xx:xx:xx:xx:ED:A5' } } }
result is:
{ brand: 'Bluesound', model: 'NODE 2', mac: 'xx:xx:xx:xx:ED:A5' }

**Comment by jghaanstra** [comment]
That looks good. Where you able to add your device and if so could you test the functionality a bit and let me know the outcome?

**Comment by ferdw** [comment]
Yes I will test tomorrow, one thing I noticed during a short test is that the track is "not known"
![2018-01-29 20_21_51-telegramdesktop](https://user-images.githubusercontent.com/10448491/35529830-81922b7c-0532-11e8-9f9f-a6ee0caa197f.png)
(don't mind the artist, it's a Spotify choice :(( )

And from the log:
![2018-01-29 20_30_03-no communication between homey and bluesound issue 1 jghaanstra_com bluesou](https://user-images.githubusercontent.com/10448491/35530072-4374bc82-0533-11e8-92cd-a22bb31f5ae0.png)

**Comment by jghaanstra** [comment]
@ferdw 

The 3 title fields used in the SyncStatus xml are not consitent and the variate between stream title, artist, track and album. Sometimes the track is in a seperate field called 'name', but if not there is no way of knowing its in title1, title2 or title3. So I cant really do anything about this. See the following xmls.

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<status etag="3fc00adc2f15b58bf0a10a1313312959">
**<album>A Change of Heart</album>**
**<artist>Ellen Andrea Wang</artist>**
<canMovePlayback>true</canMovePlayback>
<canSeek>1</canSeek>
<cursor>133</cursor>
<fn>Tidal:74588359</fn>
<image>/Artwork?service=Tidal&amp;songid=Tidal%3A74588359</image>
<indexing>0</indexing>
<mid>0</mid>
<mode>1</mode>
**<name>A Change of Heart</name>**
<pid>91</pid>
<prid>0</prid>
<quality>cd</quality>
<repeat>0</repeat>
<service>Tidal</service>
<shuffle>1</shuffle>
<sid>4</sid>
<sleep></sleep>
<song>0</song>
<state>play</state>
<streamFormat>FLAC 44100/16/2</streamFormat>
<syncStat>6</syncStat>
<title1>A Change of Heart</title1>
<title2>Ellen Andrea Wang</title2>
<title3>A Change of Heart</title3>
<totlen>262</totlen>
<volume>10</volume>
<secs>32</secs>
</status>
```

![jbqsozuommdi](https://user-images.githubusercontent.com/4710062/35585702-fc434aba-05f8-11e8-9443-56d9d41e7422.png)

**Comment by jghaanstra** [comment]
This has been resolved with release 1.0.1.
